filament/backend/src/opengl/OpenGLDriver.cpp

```diff
@@ -136,7 +136,9 @@ Driver* OpenGLDriver::create(
         }
     }
 
-    OpenGLDriver* const driver = new OpenGLDriver(ec);
+    size_t defaultSize = FILAMENT_OPENGL_HANDLE_ARENA_SIZE_IN_MB * 1024U * 1024U;
+    driverConfig.handleArenaSize = std::max(driverConfig.handleArenaSize, defaultSize);
```

Comment:
The problem is that each backend has a (potentially) different default handle arena size, driven by a different #define value: 
FILAMENT_OPENGL_HANDLE_ARENA_SIZE_IN_MB = 4 FILAMENT_METAL_HANDLE_ARENA_SIZE_IN_MB = 8
FILAMENT_VULKAN_HANDLE_ARENA_SIZE_IN_MB = 8

That is also why the DriverConfig passed into createDriver is not a const.  Only the driver knows how to validate the handle arena size because only it has the default values.  CreateDriver potentially modifies the DriverConfig.handleArenaSize when it validates the value.  Engine can then look at the modified value, if it desires.  Currently, Engine stores the validated value for reference after createDriver is called.

Reply:
It doesn't seem like FEngine needs it actually. So let's not store it in FEngine there is no point.
I really don't like in/out values -- so I think we don't store the modified value in FEngine (Which we don't need), then we can keep the parameter as const.

Reply:
Based on this and our other conversations, I will take this approach:
* Engine::Config will continue to contain a parameter for the driver handle arena size and pass it through to Driver.
* The driver will validate the value, but not modify the config struct value.
* DriverConfig will be passed as a const.
* FEngine does need to store the requested driver handle arena size so it can be used later in FEngine::loop(), but I will not read back the value to discover the validated size.